Ogra/Encodable.swift

```diff
@@ -62,22 +62,32 @@ extension Optional where T: Encodable {
 		case .Some(let v): return v.encode()
 		}
 	}
+extension CollectionType where Generator.Element == (Key: StringLiteralConvertible, Value: Encodable) {
+    public func encode() -> JSON {
+        var values = [String : JSON]()
+        for (key, value) in self {
+            values[String(key)] = value.encode()
+        }
+        return .Object(values)
+    }
 }
```

Comment:
I think this is the crux of the remaining work.

My approach here isn't yet a valid replacement for the previous dictionary (non-optional) dictionary extension. I assumed this would work, but doesn't. Sharing here so we can discuss.

If that can be resolved, I think the optional version should be trivial.